util/s3/s3Helper.go

```diff
@@ -262,17 +264,56 @@ func (helper *Helper) GetPackageFilesVersion(channel string, pkgFiles string, sp
 	maxVersion = 0
 	maxVersionName = ""
 
-	prefix := fmt.Sprintf("channel/%s/%s", channel, pkgFiles)
 	svc := s3.New(helper.session)
-	input := &s3.ListObjectsInput{
+
+	// New path schema, try these first
+	// In the case of specificVersion == 0, this is much faster than the old schema.
+	// latest/{stable,beta,nightly,dev}/{RTIME}_{version}/{node,install,tools}_{channel}_{os}_{cpu_arch}_{version}.tar.gz
+	prefix := fmt.Sprintf("latest/%s", channel)
+	input := s3.ListObjectsInput{
+		Bucket:  &helper.bucket,
+		Prefix:  &prefix,
+		MaxKeys: aws.Int64(50),
+	}
+	if specificVersion != 0 {
+		input.MaxKeys = aws.Int64(1000)
+	}
+	result, err := svc.ListObjects(&input)
```

Comment:
two suggestions:
1) switch to `ListObjectsV2` since the old one most likely will be deprecated one day
2) use double loop to bypass possible 1000 keys limit, something like this:
```
var result s3.ListObjectsOutput{IsTruncated: True}
for result.IsTruncated {
    result, err = svc.ListObjectsV2(&input)
    for _, item := range result.Contents {
        ...
    }
    if result.IsTruncated {
        input.Marker = result. NextMarker
    }
}